compiler/modules/bir/verify.bal

```diff
@@ -63,19 +66,81 @@ class VerifyContext {
     function symbolToString(Symbol sym) returns string {
         return self.mod.symbolToString(self.defn.partIndex, sym);
     }
+
+    function getBlock(Label label) returns BasicBlock {
+        return self.code.blocks[label];
+    }
+
+    function isLoopHead(Label label) returns boolean {
+        return self.loopHeads.indexOf(label) != ();
+    }
+
+    function addLoopHead(Label label) {
+        self.loopHeads.push(label);
+    }
 }
 
 public function verifyFunctionCode(Module mod, FunctionDefn defn, FunctionCode code) returns Error? {
-    VerifyContext cx = new(mod, defn);
+    VerifyContext cx = new(mod, defn, code);
     foreach BasicBlock b in code.blocks {
         check verifyBasicBlock(cx, b);
     }
+    check verifyGraph(cx, code.blocks, defn.position);
 }
 
 type IntBinaryInsn IntArithmeticBinaryInsn|IntBitwiseBinaryInsn;
 
 type Error err:Semantic|err:Internal;
 
+function verifyGraph(VerifyContext vc, BasicBlock[] blocks, Position pos) returns Error? {
+    boolean[] visited = [];
+    visited[blocks.length() - 1] = false;
+    check traveseGraph(vc, blocks, 0, visited, pos);
+    if visited.indexOf(false) != () {
+        return vc.invalidErr("unreachable blocks in function code", pos);
+    }
+}
+
+function traveseGraph(VerifyContext vc, BasicBlock[] blocks, Label label, boolean[] visited, Position pos, Label[] currentPath = []) returns Error? {
+    if currentPath.indexOf(label) != () {
+        return vc.invalidErr("forward edge graph is not acyclic", pos);
+    }
```

Comment:
This check is not strong enough since we don't travel every path due to not revisiting visited blocks.
If we travel all paths, then this is valid, but then it is slower than it needs to be.